xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/query/MailEntityQueryCondition.java

```diff
@@ -35,4 +38,31 @@ public MailEntityQueryCondition(String statusEntityId, boolean reversed)
     {
         super(statusEntityId, reversed);
     }
+
+    @Override
+    public int hashCode()
+    {
+        HashCodeBuilder builder = new HashCodeBuilder();
+
+        builder.appendSuper(super.hashCode());
+
+        return builder.build();
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj instanceof MailEntityQueryCondition) {
+            EqualsBuilder builder = new EqualsBuilder();
```

Comment:
That EqualsBuilder seems quite useless since you don't add anything to super.equals.

Reply:
Indeed I can improve it, I did a quick copy/paste to make the test passing honestly.